aten/src/ATen/native/cuda/FusedSgdKernel.cu

```diff
@@ -77,15 +78,14 @@ struct FusedSgdMathFunctor {
     if (found_inf_ptr && *found_inf_ptr == 1) {
       return;
     }
-    auto tensor_loc = tl.block_to_tensor[blockIdx.x];
-    auto chunk_idx = tl.block_to_chunk[blockIdx.x];
-    auto n = tl.numel_for_tensor[tensor_loc];
+    const index_t tensor_loc = tl.block_to_tensor[blockIdx.x];
+    const index_t chunk_idx = tl.block_to_chunk[blockIdx.x];
 
     scalar_t* args[depth];
     scalar_t r_args[depth][kILP];
     const auto all_aligned{
         init_args<depth>(args, tl, chunk_idx, chunk_size, tensor_loc)};
-    n -= chunk_idx * chunk_size;
+    const index_t n = tl.numel_for_tensor[tensor_loc] - chunk_idx * chunk_size;
```

Comment:
Just for my learning: does const allow the compiler to make optimizations knowing that it's a readonly value? Like const replacement?